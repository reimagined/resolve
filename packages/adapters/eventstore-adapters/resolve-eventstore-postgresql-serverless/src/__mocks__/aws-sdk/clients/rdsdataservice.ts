const RDSDataService = jest.fn()

const promised = (): Function =>
  jest.fn().mockReturnValue({ promise: () => Promise.resolve() })

RDSDataService.prototype.executeStatement = promised()

export default RDSDataService
