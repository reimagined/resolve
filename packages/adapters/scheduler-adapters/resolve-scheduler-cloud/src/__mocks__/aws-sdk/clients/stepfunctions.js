function StepFunctions() {}

const wrap = () =>
  jest.fn().mockReturnValue({ promise: () => Promise.resolve() })

StepFunctions.prototype.startExecution = wrap()
StepFunctions.prototype.stopExecution = wrap()
StepFunctions.prototype.listExecutions = wrap()

export default StepFunctions
