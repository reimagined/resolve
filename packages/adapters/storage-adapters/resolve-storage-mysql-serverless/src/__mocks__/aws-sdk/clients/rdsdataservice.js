const result = []

const promise = jest.fn().mockReturnValue(
  Promise.resolve({
    sqlStatementResults: [
      {
        resultFrame: {
          records: [
            {
              values: [
                { intValue: 10 },
                { bigIntValue: 1000 },
                { stringValue: '{}' }
              ]
            }
          ],
          resultSetMetadata: {
            columnCount: 3,
            columnMetadata: [
              { name: 'aggregateVersion' },
              { name: 'eventId' },
              { name: 'payload' }
            ]
          }
        }
      }
    ]
  })
)

const rdsDataService = {
  executeSql: jest.fn().mockImplementation((...args) => {
    result.push('executeSql', ...args)
    return { promise }
  })
}

const RDSDataService = function() {
  return rdsDataService
}

export default RDSDataService

export { result }
