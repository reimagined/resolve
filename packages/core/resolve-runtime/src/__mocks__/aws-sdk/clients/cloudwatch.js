const _CloudWatch = jest.fn().mockImplementation(() => CloudWatch)
const CloudWatch = _CloudWatch.bind(null)
CloudWatch.putMetricData = jest
  .fn()
  .mockReturnValue({ promise: () => Promise.resolve() })

Object.setPrototypeOf(CloudWatch, _CloudWatch)

export default CloudWatch
