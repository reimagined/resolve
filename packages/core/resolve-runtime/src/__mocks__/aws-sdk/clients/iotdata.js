export const result = []

const IotData = jest.fn().mockImplementation(function (...args) {
  result.push(['IotData constructor', ...args])
  this._publishPromiseResult = Promise.resolve()
})

IotData.prototype._toPublishPromise = jest
  .fn()
  .mockImplementation(function (...args) {
    result.push(['IotData publish.promise', ...args])

    return this._publishPromiseResult
  })

IotData.prototype.publish = jest.fn().mockImplementation(function (...args) {
  result.push(['IotData publish', ...args])

  return {
    promise: this._toPublishPromise.bind(this),
  }
})

export default IotData
