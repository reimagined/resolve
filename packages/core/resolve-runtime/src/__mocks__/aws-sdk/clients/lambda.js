let lambdaResult = null
export function setLambdaResult(value) {
  lambdaResult = value
}

export const result = []

const Lambda = jest.fn().mockImplementation(function (...args) {
  result.push(['Lambda constructor', ...args])
  this._invokePromiseResult = Promise.resolve({
    FunctionError: null,
    Payload: JSON.stringify(lambdaResult),
  })
})

Lambda.prototype._toInvokePromise = jest
  .fn()
  .mockImplementation(function (...args) {
    result.push(['Lambda invoke.promise', ...args])

    return this._invokePromiseResult
  })

Lambda.prototype.invoke = jest.fn().mockImplementation(function (...args) {
  result.push(['Lambda invoke', ...args])

  return {
    promise: this._toInvokePromise.bind(this),
  }
})

export default Lambda
