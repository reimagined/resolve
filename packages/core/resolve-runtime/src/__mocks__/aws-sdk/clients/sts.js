// export const result = []

const _STS = jest.fn().mockImplementation(() => STS)
const STS = _STS.bind(null)

STS.getCallerIdentity = jest.fn().mockReturnValue({
  promise: jest.fn().mockReturnValue({ Arn: 'RoleValidationArn' }),
})
STS.assumeRole = jest.fn()
STS.invoke = jest.fn()

Object.setPrototypeOf(STS, _STS)

// const STS = jest.fn().mockImplementation(function(...args) {
//   result.push(['STS constructor', ...args])
//   this._assumeRolePromiseResult = Promise.resolve()
// })
//
// STS.prototype._toAssumeRolePromise = jest
//   .fn()
//   .mockImplementation(function(...args) {
//     result.push(['STS assumeRole.promise', ...args])
//
//     return this._assumeRolePromiseResult
//   })
//
// STS.prototype.invoke = jest.fn().mockImplementation(function(...args) {
//   result.push(['STS invoke', ...args])
//
//   return {
//     promise: this._toAssumeRolePromise.bind(this)
//   }
// })

export default STS
