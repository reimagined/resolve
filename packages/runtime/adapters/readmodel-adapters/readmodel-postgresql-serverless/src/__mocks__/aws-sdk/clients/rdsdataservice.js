const result = []

const executeStatementPromise = jest.fn()

const resetExecuteStatementPromise = () => {
  executeStatementPromise.mockReturnValueOnce(
    Promise.resolve({
      records: [],
      columnMetadata: [],
    })
  )
}

resetExecuteStatementPromise()

const beginTransactionPromise = jest
  .fn()
  .mockReturnValue(Promise.resolve({ transactionId: 'transactionId' }))
const commitTransactionPromise = jest
  .fn()
  .mockReturnValue(Promise.resolve({ transactionStatus: 'OK' }))
const rollbackTransactionPromise = jest
  .fn()
  .mockReturnValue(Promise.resolve({ transactionStatus: 'OK' }))

const rdsDataService = {
  beginTransaction: jest.fn().mockImplementation((...args) => {
    result.push('beginTransaction', ...args)
    resetExecuteStatementPromise()
    return { promise: beginTransactionPromise }
  }),
  commitTransaction: jest.fn().mockImplementation((...args) => {
    result.push('commitTransaction', ...args)
    resetExecuteStatementPromise()
    return { promise: commitTransactionPromise }
  }),
  rollbackTransaction: jest.fn().mockImplementation((...args) => {
    result.push('rollbackTransaction', ...args)
    resetExecuteStatementPromise()
    return { promise: rollbackTransactionPromise }
  }),
  executeStatement: jest.fn().mockImplementation((...args) => {
    result.push('executeStatement', ...args)
    resetExecuteStatementPromise()
    return { promise: executeStatementPromise }
  }),
}

const RDSDataService = function () {
  return rdsDataService
}

export default RDSDataService

export { result }
