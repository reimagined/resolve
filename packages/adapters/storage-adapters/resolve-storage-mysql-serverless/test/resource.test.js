import { result as mockResult } from 'aws-sdk/clients/rdsdataservice'
import _createAdapter from '../src/index'
import createResource from '../src/resource/create'
import disposeResource from '../src/resource/dispose'
import destroyResource from '../src/resource/destroy'

describe('resource', () => {
  afterEach(() => {
    mockResult.length = 0
  })

  test('method "create" should create resource', async () => {
    const createAdapter = jest
      .fn()
      .mockImplementation((...args) => _createAdapter(...args))
    const pool = {
      createAdapter
    }

    const options = {
      awsSecretStoreAdminArn: 'awsSecretStoreAdminArn',
      dbClusterOrInstanceArn: 'dbClusterOrInstanceArn',
      databaseName: 'databaseName',
      tableName: 'tableName',
      userLogin: 'userLogin',
      userPassword: 'userPassword'
    }

    await createResource(pool, options)

    expect(createAdapter).toHaveBeenCalledWith({
      awsSecretStoreArn: options.awsSecretStoreAdminArn,
      dbClusterOrInstanceArn: options.dbClusterOrInstanceArn,
      databaseName: 'mysql',
      resourceOptions: {
        databaseName: options.databaseName,
        tableName: options.tableName,
        userLogin: options.userLogin,
        userPassword: options.userPassword
      },
      skipInit: true
    })

    expect(mockResult).toMatchSnapshot()
  })

  test('method "destroy" should destroy resource', async () => {
    const createAdapter = jest
      .fn()
      .mockImplementation((...args) => _createAdapter(...args))
    const pool = {
      createAdapter
    }

    const options = {
      awsSecretStoreAdminArn: 'awsSecretStoreAdminArn',
      dbClusterOrInstanceArn: 'dbClusterOrInstanceArn',
      databaseName: 'databaseName',
      userLogin: 'userLogin'
    }

    await destroyResource(pool, options)

    expect(createAdapter).toHaveBeenCalledWith({
      awsSecretStoreArn: options.awsSecretStoreAdminArn,
      dbClusterOrInstanceArn: options.dbClusterOrInstanceArn,
      databaseName: 'mysql',
      resourceOptions: {
        databaseName: options.databaseName,
        userLogin: options.userLogin
      },
      skipInit: true
    })

    expect(mockResult).toMatchSnapshot()
  })

  test('method "dispose" should dispose resource', async () => {
    const createAdapter = jest
      .fn()
      .mockImplementation((...args) => _createAdapter(...args))
    const pool = {
      createAdapter
    }

    Object.assign(pool, {
      createResource: createResource.bind(null, pool),
      destroyResource: destroyResource.bind(null, pool)
    })

    const options = {
      awsSecretStoreAdminArn: 'awsSecretStoreAdminArn',
      dbClusterOrInstanceArn: 'dbClusterOrInstanceArn',
      databaseName: 'databaseName',
      tableName: 'tableName',
      userLogin: 'userLogin',
      userPassword: 'userPassword'
    }

    await disposeResource(pool, options)

    expect(createAdapter).toHaveBeenCalledWith({
      awsSecretStoreArn: options.awsSecretStoreAdminArn,
      dbClusterOrInstanceArn: options.dbClusterOrInstanceArn,
      databaseName: 'mysql',
      resourceOptions: {
        databaseName: options.databaseName,
        tableName: options.tableName,
        userLogin: options.userLogin,
        userPassword: options.userPassword
      },
      skipInit: true
    })

    expect(createAdapter).toHaveBeenCalledWith({
      awsSecretStoreArn: options.awsSecretStoreAdminArn,
      dbClusterOrInstanceArn: options.dbClusterOrInstanceArn,
      databaseName: 'mysql',
      resourceOptions: {
        databaseName: options.databaseName,
        userLogin: options.userLogin
      },
      skipInit: true
    })

    expect(mockResult).toMatchSnapshot()
  })
})
