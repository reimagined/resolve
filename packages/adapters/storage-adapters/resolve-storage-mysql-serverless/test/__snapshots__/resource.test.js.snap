// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resource method "create" should create resource 1`] = `
Array [
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE DATABASE \`databaseName\`",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE USER \\"userLogin\\" IDENTIFIED BY \\"userPassword\\"",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "GRANT ALL ON \`databaseName\`.* TO \\"userLogin\\"@'%'",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE TABLE IF NOT EXISTS \`databaseName\`.\`tableName\`(
      \`eventId\` BIGINT NOT NULL AUTO_INCREMENT,
      \`timestamp\` BIGINT NOT NULL,
      \`aggregateId\` VARCHAR(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
      \`aggregateVersion\` BIGINT NOT NULL,
      \`type\` VARCHAR(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
      \`payload\` LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NULL,
      PRIMARY KEY(\`eventId\`),
      INDEX USING BTREE(\`aggregateId\`, \`aggregateVersion\`),
      INDEX USING BTREE(\`aggregateId\`),
      INDEX USING BTREE(\`aggregateVersion\`),
      INDEX USING BTREE(\`type\`),
      INDEX USING BTREE(\`timestamp\`),
      UNIQUE(\`aggregateId\`, \`aggregateVersion\`)
    )",
  },
]
`;

exports[`resource method "destroy" should destroy resource 1`] = `
Array [
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "DROP DATABASE \`databaseName\`",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "DROP USER \\"userLogin\\"",
  },
]
`;

exports[`resource method "dispose" should dispose resource 1`] = `
Array [
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "DROP DATABASE \`databaseName\`",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "DROP USER \\"userLogin\\"",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE DATABASE \`databaseName\`",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE USER \\"userLogin\\" IDENTIFIED BY \\"userPassword\\"",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "GRANT ALL ON \`databaseName\`.* TO \\"userLogin\\"@'%'",
  },
  "executeSql",
  Object {
    "awsSecretStoreArn": "awsSecretStoreAdminArn",
    "database": "mysql",
    "dbClusterOrInstanceArn": "dbClusterOrInstanceArn",
    "sqlStatements": "CREATE TABLE IF NOT EXISTS \`databaseName\`.\`tableName\`(
      \`eventId\` BIGINT NOT NULL AUTO_INCREMENT,
      \`timestamp\` BIGINT NOT NULL,
      \`aggregateId\` VARCHAR(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
      \`aggregateVersion\` BIGINT NOT NULL,
      \`type\` VARCHAR(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
      \`payload\` LONGTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NULL,
      PRIMARY KEY(\`eventId\`),
      INDEX USING BTREE(\`aggregateId\`, \`aggregateVersion\`),
      INDEX USING BTREE(\`aggregateId\`),
      INDEX USING BTREE(\`aggregateVersion\`),
      INDEX USING BTREE(\`type\`),
      INDEX USING BTREE(\`timestamp\`),
      UNIQUE(\`aggregateId\`, \`aggregateVersion\`)
    )",
  },
]
`;
